Library/Homebrew/api.rb

```diff
@@ -42,19 +42,31 @@ def fetch_json_api_file(endpoint, target:)
       retry_count = 0
       url = "#{Homebrew::EnvConfig.api_domain}/#{endpoint}"
       default_url = "#{HOMEBREW_API_DEFAULT_DOMAIN}/#{endpoint}"
+
+      # TODO: consider using more of Utils::Curl
       curl_args = %W[
         --compressed
         --speed-limit #{ENV.fetch("HOMEBREW_CURL_SPEED_LIMIT")}
         --speed-time #{ENV.fetch("HOMEBREW_CURL_SPEED_TIME")}
       ]
-      curl_args.prepend("--silent") unless Context.current.debug?
+      curl_args << "--progress-bar" unless Context.current.verbose?
+      curl_args << "--verbose" if Homebrew::EnvConfig.curl_verbose?
+      curl_args << "--silent" unless $stdout.tty?
+
+      skip_download = target.exist? &&
+                      !target.empty? &&
+                      (Homebrew::EnvConfig.no_auto_update? ||
+                      ((Time.now - Homebrew::EnvConfig.api_auto_update_secs.to_i) < target.mtime))
```

Comment:
```suggestion
                      (Time.now < (target.mtime + Homebrew::EnvConfig.api_auto_update_secs.to_i)))
```
Bit easier to read for me, but could be a matter of taste.

Reply:
@carlocab Sorry, missed this, agreed, will follow up.